gallery/visualtests/nxDrawer.js

```diff
@@ -0,0 +1,168 @@
+  }
+
+  function simpleDrawerTest(drawerId, buttonId) {
+    return async function() {
+      const drawer = await openDrawer(drawerId, buttonId);
+
+      await checkScreenshot(drawer);
+    };
+  }
+
+  function drawerA11yTest(drawerId, buttonId) {
+    return async function() {
+      await openDrawer(drawerId, buttonId);
+      await a11yTest();
+    };
+  }
+
+  beforeEach(async function() {
+    await getPage().setViewport({ width: 1366, height: 1000 });
+  });
+
+  describe('Simple NxDrawer', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-simple', 'nx-drawer-simple-open-button'));
+
+    it('passes a11y checks', drawerA11yTest('nx-drawer-simple', 'nx-drawer-simple-open-button'));
+  });
+
+  describe('Narrow NxDrawer', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-narrow', 'nx-drawer-narrow-open-button'));
+  });
+
+  describe('NxDrawer With Footer', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-footer', 'nx-drawer-with-footer-open-button'));
+  });
+
+  describe('NxDrawer With Subtitle', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-subtitle', 'nx-drawer-with-subtitle-open-button'));
+  });
+
+  describe('NxDrawer With Paragraph', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-paragraph', 'nx-drawer-with-paragraph-open-button'));
+  });
+
+  describe('NxDrawer With Subtitle and Paragraph', function() {
+    it('looks right', simpleDrawerTest('nx-drawer-with-subtitle-and-paragraph',
+        'nx-drawer-with-subtitle-and-paragraph-open-button'));
+  });
+
+  describe('NxDrawer + NxDropdown ESC Closing Behavior', function() {
+    async function isFocused(el) {
+      return el.evaluate(e => e === document.activeElement);
+    }
+
+    async function isInDocument(el) {
+      return el.evaluate(e => e.isConnected);
+    }
+
+    async function pressEsc() {
+      const { keyboard } = getPage();
+
+      await keyboard.press('Escape');
+    }
+
+    it('closes one layer per ESC press', async function() {
+      const [openButton] = await waitAndGetElements('#nx-drawer-esc-open-button');
+
+      await openButton.click();
+
+      await wait(200);
+
+      const [drawer, dropdownToggle] = await waitAndGetElements(
+          '#nx-drawer-esc',
+          '#nx-drawer-esc .nx-dropdown__toggle'
+      );
+
+      expect(await isInDocument(drawer)).toBe(true);
+      expect(await isInDocument(dropdownToggle)).toBe(true);
+
+      await dropdownToggle.click();
+
+      const [dropdownMenu] = await waitAndGetElements('#nx-drawer-esc .nx-dropdown-menu');
+
+      expect(await isInDocument(dropdownMenu)).toBe(true);
+
+      await pressEsc();
+
+      expect(await isInDocument(dropdownMenu)).toBe(false);
+
+      await pressEsc();
+
+      await wait(200);
+
+      expect(await isInDocument(drawer)).toBe(false);
+
+      expect(await isFocused(openButton)).toBe(true);
+    });
+  });
+
+  describe('NxDrawer With Global Header', function() {
```

Comment:
I'd say that some of these tests supercede some of the others, since in addition to showing the overall alignment they should drawers with all possible content.  IMO we only need one snapshot of each combination of drawer features/contents.

Some of these screenshots I'd say can overlap in purpose.  For instance the footer and the header are far enough apart that we can reasonably assume that they won't interact with each other, and so we don't need every permutation of header contents both with and without the footer.  On the other hand, the header contents are all right next to each other  an affect each other and so I _do_ think we need to test every permutation there: title only, title & subtitle, title & description, title & subtitle & description.

But we don't need to test all of that up close and then also separately in the global sidebar layout.  Any of the above that are covered in the global sidebar screenshots are covered. No need to test them again separately.

Reply:
> I'd say that some of these tests supercede some of the others, since in addition to showing the overall alignment they should drawers with all possible content. IMO we only need one snapshot of each combination of drawer features/contents.
> 
> Some of these screenshots I'd say can overlap in purpose. For instance the footer and the header are far enough apart that we can reasonably assume that they won't interact with each other, and so we don't need every permutation of header contents both with and without the footer. On the other hand, the header contents are all right next to each other an affect each other and so I _do_ think we need to test every permutation there: title only, title & subtitle, title & description, title & subtitle & description.
> 
> But we don't need to test all of that up close and then also separately in the global sidebar layout. Any of the above that are covered in the global sidebar screenshots are covered. No need to test them again separately.

Updated tests here: 8356f233f575973b8be5be477b77d35da0b492c6